tests/iio_writedev.c

```diff
@@ -457,8 +457,11 @@ int main(int argc, char **argv)
 
 			if (num_samples) {
 				num_samples -= write_len / sample_size;
-				if (!num_samples)
-					quit_all(EXIT_SUCCESS);
+				if (!num_samples) {
+					exit_code = EXIT_SUCCESS;
+					if (!cyclic_buffer)
```

Comment:
If it is, then you should be able to do `if(!num_samples && !cyclic_buffer)` correct?

Reply:
there's various smarter ways to do this;
this was one that is a simple-looking one;

another one would be `app_running = cyclic_buffer`
either works;
i don't care much which we use as long as it's simple looking enough, so that nobody asks anything about it;
the combination of 2 bool vars can look slightly confusing in following code-path